src/utils/WalletUtils.ts

```diff
@@ -142,3 +142,58 @@ export async function delegate(
         throw error
     }
 }
+
+
+
+export async function withdraw(
+    walletFileContents: string,
+    passphrase: string,
+    contractAddress: string,
+    amount: number,
+    tezosNodeAddress: string
+): Promise<string> {
+
+    const { privateKey, address } = await unlockWallet(walletFileContents, passphrase);
+
+    const tezos = new TezosToolkit(tezosNodeAddress);
+
+    tezos.setProvider({
+        signer: new InMemorySigner(privateKey)
+    });
+    const contract1 = await tezos.contract.at(contractAddress);
+    console.log({tezos})
+    console.log({contract1})
+
+    try {
+        // Withdraw the funds
+        const contract = await tezos.contract.at(contractAddress);
+        const op = await contract.methods.transfer({
+            to: address,
+            amount: amount // Amount is in tez (1 tez = 1,000,000 microtez)
+        }).send();
+
+        // Confirm the transaction
+        const confirmation = await op.confirmation(1);
+        console.log(`Transaction successfully included in block: ${confirmation}`);
+        console.log(`Operation hash: ${op.opHash}`);
+        return op.opHash;
+    } catch (error) {
+        console.error(`Error in withdrawing tez: ${error}`);
+        throw error
+    }
+}
+
+export async function getDelegatorContracts(tz1Address: string) {
+    try {
+        const response = await fetch(`https://api.tzkt.io/v1/accounts/${tz1Address}/contracts`);
```

Comment:
Please see https://api.tzkt.io/#operation/Accounts_GetByAddress.

If we only want to get KT1 delegation contracts then we should filter for kind `delegator_contract` (as opposed to `smart_contract`).